packages/client-core/src/user/components/UserMenu/menus/ProfileMenu.tsx

```diff
@@ -384,7 +392,7 @@ const ProfileMenu = ({ className, hideLogin, isPopover, changeActiveMenu, onClos
         <section className={styles.profileBlock}>
           <div className={styles.avatarBlock}>
             <img src={getAvatarURLForUser(userAvatarDetails, userId)} alt="" crossOrigin="anonymous" />
-            {changeActiveMenu && (
+            {allowAvatarChange && changeActiveMenu && (
```

Comment:
rather than passing down allowAvatarChange we can enable this only when `getState(EngineState).joinedWorld.value` is true

Reply:
I passed the flag, so that if at later point we need to allow other pages, i.e. admin panel to allow avatar change. Then all we have to do it to handle `Views.AvatarChange` in `changeActiveMenu` listener and it can allow us to change avatar.

Though if you want I can make it work with `getState(EngineState).sceneLoaded.value` and remove `allowAvatarChange`

Reply:
okay, leave as is for now, we can impove it in the future